ibc/src/component/client.rs

```diff
@@ -748,26 +748,31 @@ mod tests {
             anchor: tct::Tree::new().root(),
         };
 
-        Ics2Client::check_tx_stateless(&create_client_tx).unwrap();
+        let ctx = Context::new();
+        Ics2Client::check_tx_stateless(ctx.clone(), &create_client_tx).unwrap();
         client_component
-            .check_tx_stateful(&create_client_tx)
+            .check_tx_stateful(ctx.clone(), &create_client_tx)
             .await
             .unwrap();
         // execute (save client)
-        client_component.execute_tx(&create_client_tx).await;
+        client_component
+            .execute_tx(ctx.clone(), &create_client_tx)
+            .await;
 
         assert_eq!(client_component.state.client_counter().await.unwrap().0, 1);
 
         // now try update client
 
-        Ics2Client::check_tx_stateless(&update_client_tx).unwrap();
+        Ics2Client::check_tx_stateless(ctx.clone(), &update_client_tx).unwrap();
         // verify the ClientUpdate proof
         client_component
-            .check_tx_stateful(&update_client_tx)
+            .check_tx_stateful(ctx.clone(), &update_client_tx)
```

Comment:
it occurred to me that the new Context will be useful for testing components (can assert against events etc)